src-tauri/src/lib.rs

```diff
@@ -13,67 +15,163 @@ pub struct ChangedRow {
     pub changes: HashMap<String, ChangeValue>,
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct AddedOrRemovedRow {
+    pub id: String,
+    pub record: HashMap<String, String>,
+}
+
 #[derive(Serialize, Deserialize)]
 #[serde(untagged)]
 pub enum ChangeValue {
     Single(String),
-    Pair(Vec<String>),
+    Pair(Vec<String>), // [old_value, new_value]
 }
 
 pub fn compare_csv_contents(
     file1_content: String,
     file2_content: String,
     comparison_header: String,
 ) -> Result<ComparisonResult, String> {
+    // Normalize the comparison header to remove any extra quotes or whitespace
+    let normalized_comparison_header = comparison_header.trim_matches('\"').trim().to_string();
```

Comment:
This is to address certain headers that are empty, have a quote, or are just hanging around with no values